corehq/apps/export/views.py

```diff
@@ -1569,10 +1569,22 @@ class DataFileDownloadDetail(BaseProjectDataView):
     urlname = 'download_data_file'
 
     def get(self, request, *args, **kwargs):
+
+        def blob_iterator(blob_):
+            chunk_size = 1000
+            while True:
+                chunk = blob_.read(chunk_size)
+                if not len(chunk):
+                    break
+                yield chunk
+
+        def get_blob_iterator(blob_):
+            return blob_ if hasattr(blob_, '__iter__') else blob_iterator(blob_)
+
         try:
             data_file = DataFile.objects.filter(domain=self.domain).get(pk=kwargs['pk'])
             blob = data_file.get_blob()
-            response = StreamingHttpResponse(blob, content_type=data_file.content_type)
+            response = StreamingHttpResponse(get_blob_iterator(blob), content_type=data_file.content_type)
```

Comment:
here's an example of a slightly easier way to do what you're doing in case useful: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/soil/__init__.py#L297